packages/core/src/lib/lit-core.ts

```diff
@@ -223,41 +224,66 @@ export class LitCore {
     return globalThis.logManager.getLogsForId(id);
   };
 
-  private async _getValidatorData() {
-    const [minNodeCount, bootstrapUrls] = await Promise.all([
-      LitContracts.getMinNodeCount(
-        this.config.litNetwork,
-        this.config.contractContext,
-        this.config.rpcUrl
-      ),
-      LitContracts.getValidators(
-        this.config.litNetwork,
-        this.config.contractContext,
-        this.config.rpcUrl,
-        this.config.nodeProtocol
-      ),
-    ]);
+  /**
+   * Retrieves the validator data including staking contract, epoch, minNodeCount, and bootstrapUrls.
+   * @returns An object containing the validator data.
+   * @throws Error if minNodeCount is not provided, is less than or equal to 0, or if bootstrapUrls are not available.
+   */
+  private async _getValidatorData(): Promise<{
+    stakingContract: ethers.Contract;
+    epochInfo: EpochInfo;
+    minNodeCount: number;
+    bootstrapUrls: string[];
+  }> {
+    const { stakingContract, epochInfo, minNodeCount, bootstrapUrls } =
+      await LitContracts.getConnectionInfo({
+        litNetwork: this.config.litNetwork,
+        networkContext: this.config.contractContext,
+        rpcUrl: this.config.rpcUrl,
+        nodeProtocol: this.config.nodeProtocol,
+      });
 
-    if (minNodeCount <= 0) {
+    // Validate minNodeCount
+    if (minNodeCount === undefined || minNodeCount === null) {
+      throw new Error('minNodeCount is required');
```

Comment:
```suggestion
  throw new Error('Could not obtain minNodeCount');
```

"required" gives me the impression that the consumer has to provide it

Reply:
Good point!